app/src/main/kotlin/com/antosik/benchproject/app/favourite/movies/viewModel/FavouriteMoviesViewModel.kt

```diff
@@ -0,0 +1,26 @@
+package com.antosik.benchproject.app.favourite.movies.viewModel
+
+import androidx.lifecycle.ViewModel
+import com.antosik.benchproject.app.common.state.UIState.Empty
+import com.antosik.benchproject.app.common.state.UIState.Success
+import com.antosik.benchproject.app.popular.movies.entity.toUi
+import com.antosik.benchproject.domain.favourite.usecase.GetFavouriteMoviesUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.flow
+
+@HiltViewModel
+class FavouriteMoviesViewModel @Inject constructor(
+    private val getFavouriteMoviesUseCase: GetFavouriteMoviesUseCase,
+) : ViewModel() {
+
+    val favouriteMoviesViewState = flow {
```

Comment:
just state will be enough in my opinion :)

Reply:
What do you mean just state? Do you think about creating _StateFlow_ or something else? 😄

Reply:
I tried to do something like this: 

FavouriteMoviesViewModel.kt

```
    private val _favouriteMoviesViewState = MutableStateFlow<UIState>(Loading)
    val favouriteMoviesViewState = _favouriteMoviesViewState.asStateFlow()

    init {
        loadFavouriteMovies()
    }

    private fun loadFavouriteMovies() {
        viewModelScope.launch {
            getFavouriteMoviesUseCase().collect { moviesModel ->
                if (moviesModel.isNotEmpty()) {
                    _favouriteMoviesViewState.value = Success(moviesModel.toUi())
                } else {
                    _favouriteMoviesViewState.value = Empty
                }
            }
        }
    }
```

And my UI

```
    @SuppressLint("StateFlowValueCalledInComposition")
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?,
    ) = ComposeView(requireContext()).apply {
        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
        setContent {
            BenchProjectTheme {
                Column(
                    modifier = Modifier
                        .verticalScroll(rememberScrollState())
                        .padding(bottom = 58.dp),
                ) {
                    val favouriteMoviesState = favouriteMoviesViewModel.favouriteMoviesViewState
                    when (favouriteMoviesState.value) {
                        is Loading -> {
                            // TODO
                        }

                        is Success<*> -> {
                            Text(
                                text = favouriteMoviesState.value.data<List<Movie>>().toString(),
                                modifier = Modifier
                                    .fillMaxWidth()
                            )
                        }

                        is Empty -> {
                            // TODO
                        }
                    }
                }
            }
        }
    }
```

records from DB don't update. When I first enter the UI I see nothing, after re-entering, and only then can see something. When I added something to favourite movies my list hasn't updated.

Reply:
Or if you think of just State I tried to do something like this:
**FavouriteMoviesViewModel.kt**
```
    private val _favouriteMoviesViewState = mutableStateOf<UIState>(Loading)
    val favouriteMoviesViewState: State<UIState> = _favouriteMoviesViewState

    init {
        loadFavouriteMovies()
    }

    private fun loadFavouriteMovies() {
        viewModelScope.launch {
            getFavouriteMoviesUseCase().collect { moviesModel ->
                if (moviesModel.isNotEmpty()) {
                    _favouriteMoviesViewState.value = Success(moviesModel.toUi())
                } else {
                    _favouriteMoviesViewState.value = Empty
                }
            }
        }
    }
```
And my UI

```
        setContent {
            BenchProjectTheme {
                Column(
                    modifier = Modifier
                        .verticalScroll(rememberScrollState())
                        .padding(bottom = 58.dp),
                ) {
                    val favouriteMoviesState by favouriteMoviesViewModel.favouriteMoviesViewState
                    when (favouriteMoviesState) {
                        is Loading -> {
                            // TODO
                        }

                        is Success<*> -> {
                            Text(
                                text = favouriteMoviesState.data<List<Movie>>().toString(),
                                modifier = Modifier
                                    .fillMaxWidth()
                            )
                        }

                        is Empty -> {
                            // TODO
                        }
                    }
                }
            }
        }
```

but also my UI didn't update

Reply:
Ok, now it's working when I called the method from UI I uploaded my changes, can you check these 😄

Reply:
Done 😄